apps/asap-server/test/handlers/teams/index-handler.test.ts

```diff
@@ -0,0 +1,50 @@
+import { EventBridgeEvent } from 'aws-lambda';
+import {
+  indexHandlerFactory,
+  SquidexWebhookTeamPayload,
+} from '../../../src/handlers/teams/index-handler';
+import { TeamsEventType } from '../../../src/handlers/webhooks/webhook-teams';
+import { createEventBridgeEventMock } from '../../helpers/events';
+import {
+  algoliaClientMock,
+  algoliaIndexMock,
+} from '../../mocks/algolia-client.mock';
+import { teamControllerMock } from '../../mocks/team-controller.mock';
+import { teamResponse } from '../../fixtures/teams.fixtures';
+
+describe('Team Research Outputs Index', () => {
+  const indexHandler = indexHandlerFactory(
+    teamControllerMock,
+    algoliaClientMock,
+  );
+
+  afterEach(() => jest.clearAllMocks());
+
+  test('Should fetch the team and create a record in Algolia for every research output', async () => {
+    teamControllerMock.fetchById.mockResolvedValueOnce(teamResponse);
+
+    await indexHandler(getEvent());
+
+    expect(algoliaIndexMock.saveObjects).toHaveBeenCalledWith(
+      teamResponse.outputs.map((output) => ({
```

Comment:
not a huge fan of such logic in tests -especially since you are effectively coping it from the implementing function. 

I would probably reassign `teamResponse.outputs` in the test and then expect these exact objects that you define.